dagfactory/dagbuilder.py

```diff
@@ -396,21 +402,27 @@ def build(self) -> Dict[str, Union[str, DAG]]:
                     task_conf["task_group"] = task_groups_dict[
                         task_conf.get("task_group_name")
                     ]
+                params: Dict[str, Any] = {
+                    k: v for k, v in task_conf.items() if k not in SYSTEM_PARAMS
+                }
+
                 task: BaseOperator = DagBuilder.make_task(
                     operator=operator, task_params=params
                 )
+
                 tasks_dict[task.task_id]: BaseOperator = task
+
             elif "generator" in task_conf:
+
                 # Task generator
-                generator: str = task_conf[
-                    "generator"
-                ]  # dagfactory.generators.airbyte_dbt.AirbyteDbtGenerator
+                # dagfactory.generators.airbyte_dbt.AirbyteDbtGenerator
+                generator: str = task_conf["generator"]
                 class_name = generator.split(".")[-1]
                 module_path = generator.replace(f".{class_name}", "")
                 module = importlib.import_module(module_path)
                 instance = getattr(module, class_name)(self)
-                tasks = instance.generate_tasks(params)
-                tasks_dict.update(tasks)
+                tasks_dict = instance.generate_tasks(dag)
```

Comment:
Instead of replacing completely the tasks_dict, you should update it, so new tasks coexist with existing ones.